crates/typescript-tests/src/ts_generic_sig.rs

```diff
@@ -0,0 +1,291 @@
+#![allow(clippy::all)]
+
+use serde::{Deserialize, Serialize};
+use wasm_bindgen::__rt::*;
+use wasm_bindgen::convert::*;
+use wasm_bindgen::describe::*;
+use wasm_bindgen::prelude::*;
+use wasm_bindgen::{JsCast, JsValue};
+
+#[derive(Serialize, Deserialize)]
+pub struct SomeType {
+    pub prop: String,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct SomeGenericType<T> {
+    pub field: T,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct OtherGenericType<T, E> {
+    pub field1: T,
+    pub field2: E,
+}
+
+// some error type
+pub struct Error;
+
+#[wasm_bindgen]
+pub struct TestStruct;
```

Comment:
All types here seem to be _exported_ types. Can we extend the coverage here to imported types as well?

We would also need to cover interactions with `js_name` and `js_namespace` renaming to ensure aliasing is supported.